Mage.Server.Plugins/Mage.Player.Human/src/mage/player/human/HumanPlayer.java

```diff
@@ -872,21 +904,38 @@ public boolean chooseTargetAmount(Outcome outcome, TargetAmount target, Ability
                 required = false;
             }
 
-            updateGameStatePriority("chooseTargetAmount", game);
-            prepareForResponse(game);
-            if (!isExecutingMacro()) {
-                // target amount uses for damage only, if you see another use case then message must be changed here and on getMultiAmount call
-                String message = String.format("Select targets to distribute %d damage (selected %d)", amountTotal, target.getTargets().size());
-                game.fireSelectTargetEvent(playerId, new MessageToClient(message, getRelatedObjectName(source, game)), possibleTargets, required, options);
+            UUID responseId = null;
+            boolean canAutochoose = target.getMinNumberOfTargets() == target.getMaxNumberOfTargets() && // Targets must be picked
+                    target.getNumberOfTargets() - target.getSize() == possibleTargetIds.size(); // Available targets are equal to the number that must be picked
+            if (canAutochoose) {
+                // Targets can be auto-chosen
+                for (UUID possibleTargetId : possibleTargetIds) {
+                    if (!target.getTargets().contains(possibleTargetId)) {
+                        // Make sure to not autopick one that's already been picked, otherwise we may hit an infinite loop
+                        responseId = possibleTargetId;
+                        break;
+                    }
+                }
+            }
+
+            if (responseId == null) {
+                updateGameStatePriority("chooseTargetAmount", game);
+                prepareForResponse(game);
+                if (!isExecutingMacro()) {
+                    // target amount uses for damage only, if you see another use case then message must be changed here and on getMultiAmount call
+                    String message = String.format("Select targets to distribute %d damage (selected %d)", amountTotal, target.getTargets().size());
+                    game.fireSelectTargetEvent(playerId, new MessageToClient(message, getRelatedObjectName(source, game)), possibleTargetIds, required, options);
+                }
+                waitForResponse(game);
+
+                responseId = getFixedResponseUUID(game);
```

Comment:
Need to look into if this one also needs automation (beyond staying constant). Not sure if there are effects that use chooseTargetAmount that meet the criteria (i.e. any that don't use "up to").